dask/dataframe/io/parquet/arrow.py

```diff
@@ -201,25 +201,30 @@ def _collect_pyarrow_dataset_frags(
                 for name in partition_names
             ]
 
-        if split_row_groups is False and (ds_filters is None or not handle_divisions):
-            # Avoid row-group splitting.
-            # NOTE: We don't do this if we are both filtering and handling
-            # divisions, because the resulting divisions will be calculated
-            # with UNFILTERED row-groups.  This is probably fine in most cases,
-            # but could be an issue if the to-be index is not monotonically
-            # increasing before the filters are applied.
-            # TODO (maybe): Even if we filter out row-groups before aggregating statistics,
-            # we are not capturing the intra-row-group filtering that will be
-            # performed at read time.  Therefore, it may be best to use this code
-            # path for all `split_row_groups==False` cases (assuming it is generally
-            # more performant to avoid splitting).
-            if gather_statistics:
-                file_frag.ensure_complete_metadata()
-            metadata.append(file_frag)
+        # Append fragements to our "metadata" list
+        if ds_filters:
+            # If we have filters, we need to split the row groups to apply them.
+            # If any row-groups are filtered out, we convert the remaining row-groups
+            # to a NEW (filtered) fragment, and append the filtered fragment to our
+            # metadata.
+            filtered_row_groups = [
+                rg_frag.row_groups[0].id
+                for rg_frag in file_frag.split_by_row_group(
+                    ds_filters, schema=ds.schema
+                )
+            ]
+            if len(file_frag.row_groups) > len(filtered_row_groups):
+                filtered_frag = file_frag.format.make_fragment(
+                    file_frag.path,
+                    file_frag.filesystem,
+                    file_frag.partition_expression,
+                    row_groups=filtered_row_groups,
+                )
+                metadata.append(filtered_frag)
+            else:
+                metadata.append(file_frag)
         else:
-            # Loop over row-group fragments
-            for rg_frag in file_frag.split_by_row_group(ds_filters, schema=ds.schema):
-                metadata.append(rg_frag)
```

Comment:
This part you want to keep if `split_row_groups is True` ?

Reply:
Good question...

I am thinking that the most general solution is to just pass along "file" fragments and a list of row-group IDs to `read_partition`.  This way, the worker can use `make_fragment` to select one or more row-groups (depending on the `split_row_groups` setting).

At this point we have separated most of the `ArrowDatasetEngine` logic from the `ArrowLegacyEngine` logic, so we can try whatever we want.  Do you think it makes more sense to use split_by_row_group and aggregate row-group fragments in `read_partition`?

Reply:
Just from a user perspective: if you specify `split_row_groups=True`, you should get a task per row group, so we still need to split it somewhere? 
Now it is only splitted in the `if filters` code path, but also when there are no filters, the user expects one dask partition mapping to one row group?

Reply:
>you should get a task per row group, so we still need to split it somewhere?

Right, **the split now happens at read time (in `_read_table`)**.  The worker will get a **file**-level `ParquetFileFragment` (or a file path) and a list of row-group IDs.  It will effectively perform the split by using the `make_fragment` method you shared.